src/connection_state_machine.c

```diff
@@ -150,10 +152,15 @@ enum OperationReturnCode csm_next_state(struct state_machine_ctx_t *ctx)
         }
         else
         {
-            csm_set_state(ctx, LDAP_CONNECTION_STATE_RUN);
+            csm_set_state(ctx, LDAP_CONNECTION_STATE_LOAD_SCHEMA);
         }
         break;
 
+    case LDAP_CONNECTION_STATE_LOAD_SCHEMA:
+        rc = ldap_schema_load(ctx->ctx, ctx->ctx->schema);
```

Comment:
New signature of `ldap_schema_load` does not require a `schema` parameter. We need to remove it.